saged/pickle_tsv.py

```diff
@@ -12,7 +15,22 @@
                         )
     args = parser.parse_args()
 
+    dtypes = {'study': str}
+
     with open(args.tsv_path) as tsv_file:
-        df = pd.read_csv(tsv_file, sep='\t', index_col=0)
+        header = tsv_file.readline()
+        header = header.strip().split('\t')[1:]
+
+        for col in header:
+            dtypes[col] = np.half  # half precision floats
+
+    chunksize = 10000
+
+    df = pd.DataFrame()
+    data_iterator = pd.read_csv(args.tsv_path, sep='\t', index_col=0, dtype=dtypes,
+                                chunksize=chunksize)
+    for sub_data in tqdm.tqdm(data_iterator, total=32):
+        df = df.append(sub_data)
```

Comment:
Correct, the original dataset is an ASCII flat file, so each entry has a ~10 bytes representing each number in the table. The final pickled dataframe ends up only being 3GB, but I added the changes here because I was running out of memory in initial runs. 

As it turns out, moving from 60k genes to 5k genes made the data smaller, but I left this file in its more general version in case I get bigger data